source/_posts/2017-09-01-gsoc-2017-data-visualization-using-daru-view.markdown

```diff
@@ -0,0 +1,284 @@
+
+We want GoogleChart adapter to be very strong since Google chart tools always gets updated and it has amazing plotting features. Similar to the HighCharts module, here also we can use all the options described in Google Charts website.
+
+User can access the `google_visualr` object using `Daru::View::Plot#chart`, if
+they want to operate `google_visualr` methods.
+
+
+#### GoogleCharts as data table
+
+One of the good thing about googe chart is, it can be used for generating table
+for web applcation and IRuby Notebook with pagination and other features.
+
+**`Daru::View::Plot` can take data Array, daru DataFrame, Daru Vector,
+Daru::View::Table as data source.**
+
+**`Daru::View::Table` can take data Array, daru DataFrame, Daru Vector as data
+ source.**
+
+
+### DataTables
+
+[DataTables](https://datatables.net/) has interaction controls to any HTML table. It can handle large set of data and have many cool features.
+
+To use it, daru-view uses [https://github.com/Shekharrajak/data_tables](https://github.com/Shekharrajak/data_tables) gem. [Note: the gem name will be changed in near future]
+
+It basically uses the HTML table code and add features that user want. So internally HTML table code of daru DataFrame and daru Vector is passed
+as data source parameter.
+
+
+## Future Work
+
+daru-view will be more powerful and simple in near future. Developers can add
+more libraris in daru-view easily, if required.
+
+## Conclusion
+
+The aim of the daru-view is to plot charts in IRuby notebook and ruby web
+application easily, so that developers need not have to use any other gem or
+language for visualization.
+
+It can work smoothly in Rails/Sinatra/Nanoc web frameworks and I hope it can work in other ruby frameworks as well, because daru-view is generating the html code and javascript code for the chart, which is basic need of the webpage.
+
+**Why not use the plotting libraries directly?**
+
+If you are using daru gem for analysing the data and want to visualise it, then it will be good if you have data-visualisation within daru and can plot it directly using DataFrame/Vector objects of daru.
+
+daru-view will be helpful in plotting charts and tables directly from the Daru::DataFrame and Daru::Vector . daru-view using nyaplot, highcharts , google charts right now to plot the chart. So user can set the plotting library and get the chart accordingly.
+
+Most of the plotting libraries doesn't provide the features of plotting charts in iruby notebook. They are defined only for web applications (mostly for Rails). But daru-view can plot charts in any ruby web application as well as iruby notebook.
+
+## Acknowledgements
+
+I would like to thank to my mentors [Sameer Deshmukh](https://github.com/v0dro)
```

Comment:
Also add a note about the Ruby Science Foundation (with link) and Google Summer of Code 2017.

Reply:
Done.